### PR TITLE
new: Add `linode_placement_group_assignment` resource

### PR DESCRIPTION
**Pull request by lgarber-akamai**
## 📝 Description

This pull request adds a new `linode_placement_group_assignment` resource, which manages a single assignment between a Placement Group and a Linode.

Additionally, this PR adds a new `placement_group_externally_managed` attribute to the `linode_instance` resource which disables diffs on the `placement_group` field when enabled.

NOTE: Because this resource does not correspond to a single API resource, the ID attribute is stored as a base64-encoded JSON string containing the Linode ID and PG ID. 

## ✔️ How to Test

The following test steps assume you have pulled down this PR locally and pointed your local environment at an API instance where PGs are available:

```
export LINODE_URL=...
export LINODE_TOKEN=...
```

### Integration Testing

```
make PKG_NAME=linode/placementgroupassignment int-test
make PKG_NAME=linode/instance int-test
```

### Manual Testing

1. In a Terraform provider sandbox environment (e.g. dx-devenv), apply the following configuration:

```terraform
resource "linode_placement_group_assignment" "my-assignment" {
  placement_group_id = linode_placement_group.my-pg.id
  linode_id = linode_instance.my-inst.id
}

resource "linode_placement_group" "my-pg" {
  label = "my-pg"
  region = "us-east"
  affinity_type = "anti_affinity:local"
}

resource "linode_instance" "my-inst" {
  label = "my-inst"
  region = "us-east"
  type = "g6-nanode-1"
  placement_group_externally_managed = true
}
```

2. Ensure the apply finishes successfully.
3. Navigate to the newly created Placement Group in Cloud Manager and ensure the new instance has been assigned to it.
4. Remove the Placement Group assignment from the Terraform state:

```
terraform state rm linode_placement_group_assignment.my-assignment
```

5. Run the following command to re-import the previously created Placement Group:

```
make import ARGS="linode_placement_group_assignment.my-assignment MYPGID,MYINSTANCEID"
```

6. Re-apply the configuration and ensure no changes are proposed.
7. Comment out the `linode_placement_group_assignment` resource in the configuration and re-apply.
8. Ensure the apply finishes successfully.
9. Navigate to the Placement Group in Cloud Manager and ensure the Placement Group does not have any assigned Linodes.

### CODE REVIEW

File: linode/instance/schema_resource.go [hunk 1]
```diff
@@ -454,6 +454,19 @@ var resourceSchema = map[string]*schema.Schema{
 		Description: "Fields related to the Placement Group this instance is assigned to.",
 		Optional:    true,
 		MaxItems:    1,
+		DiffSuppressFunc: func(k, oldValue, newValue string, d *schema.ResourceData) bool {
+			return d.Get("placement_group_externally_managed").(bool)
+		},
+	},
+	"placement_group_externally_managed": {
```
Comment (lgarber-akamai) [inline linode/instance/schema_resource.go:461]: Users can _technically_ achieve similar functionality by adding `placement_group` to the `ignore_changes` lifecycle policy but I figured a dedicated attribute would add some extra user confidence and give us a bit more flexibility.

File: linode/instance/schema_resource.go [hunk 2]
```diff
@@ -454,6 +454,19 @@ var resourceSchema = map[string]*schema.Schema{
...
+		Type: schema.TypeBool,
+		Description: "If true, this placement group's assignment is externally managed and will " +
+			"NOT be updated by this resource.",
+		Optional: true,
+		Default:  false,
+		DiffSuppressFunc: func(k, oldValue, newValue string, d *schema.ResourceData) bool {
+			return true
```
Comment (lgarber-akamai) [inline linode/instance/schema_resource.go:468]: Changing this field shouldn't trigger any diffs.

File: linode/placementgroupassignment/framework_resource.go [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+			resp.Diagnostics.AddWarning(
+				fmt.Sprintf(
+					"Attempted to delete PG assignment to Linode %d but the resource was not found",
+					linodeID,
+				),
+				err.Error(),
+			)
+		} else if lErr, ok := err.(*linodego.Error); ok &&
+			lErr.Code == 400 &&
+			strings.Contains(lErr.Message, "does not belong to Placement Group") {
```
Comment (lgarber-akamai) [inline linode/placementgroupassignment/framework_resource.go:226]: Checking the error message here is definitely not ideal but I couldn't think of another way to differentiate between an invalid request body and an already-unassigned Linode.

File: linode/instance/schema_resource.go (hunk 1 above)
Comment (lgarber-akamai) [reply linode/instance/schema_resource.go:461]: Hmm, good question! My only concern with making the `placement_group` computed would be that users would lose the ability to unassign their instance from the PG

**Review by ykim-1 — APPROVED** [review]
I had to change the `testRegion` to `us-east` to get the tests to pass but LGTM working as expected. Excellent work!

**Comment by yec-akamai** [comment]
> I had to change the `testRegion` to `us-east` to get the tests to pass but LGTM working as expected. Excellent work!

Same for me when running the instance integration test. Should we change the testRegion in code?

**Review by yec-akamai — APPROVED** [review]
Tests passed locally, nice work!

**Pull request merged**